backend/zeno_backend/database/select.py

```diff
@@ -51,115 +102,95 @@ def models(project: str) -> list[str]:
+        params = [user.id, user.id, user.id]
+        likes_query = sql.SQL(
+            "SELECT project_uuid, COUNT(*) as total_likes"
+            " FROM project_like GROUP BY project_uuid "
+        )
+        projects_query = (
+            sql.SQL("WITH LikesSummary AS (")
+            + likes_query
+            + sql.SQL("),")
+            + sql.SQL(" CombinedProjects AS (")
+            + PROJECTS_BASE_QUERY
+            + sql.SQL(") ")
+            + sql.SQL(
+                "SELECT cp.*, COALESCE(ls.total_likes, 0) AS total_likes FROM"
+                " CombinedProjects cp LEFT JOIN LikesSummary ls "
+                " ON cp.uuid = ls.project_uuid ORDER BY total_likes DESC "
             )
         )
+        if projects_request.limit:
+            projects_query += sql.SQL(" LIMIT %s ")
+            params += [projects_request.limit, projects_request.offset]
+        else:
+            params += [projects_request.offset]
+        projects_query += sql.SQL(" OFFSET %s; ")
+        projects_result = db.execute_return(projects_query, params)
 
-        user_projects_result = db.execute_return(
-            "SELECT p.uuid, p.name, p.owner_id, p.view, "
-            "p.samples_per_page, up.editor, p.public, "
-            "p.description FROM projects AS p JOIN user_project AS up "
-            "ON p.uuid = up.project_uuid WHERE up.user_id = %s;",
-            [user.id],
-        )
-        user_projects = []
-        for res in user_projects_result:
-            owner_name = db.execute_return(
-                "SELECT name FROM users WHERE id = %s", [res[2]]
-            )
-            if len(owner_name) != 0 and owner_name[0][0] != user.name:
-                owner_name = owner_name[0][0]
-                user_projects.append(
-                    Project(
-                        uuid=res[0],
-                        name=res[1],
-                        owner_name=str(owner_name),
-                        view=match_instance_view(res[3]),
-                        samples_per_page=res[4],
-                        editor=res[5],
-                        public=res[6],
-                        description=res[7],
-                    )
-                )
-
-        project_org_result = db.execute_return(
-            "SELECT p.uuid, p.name, p.owner_id, p.view,"
-            " p.samples_per_page, op.editor, p.public, p.description "
-            "FROM projects AS p JOIN (SELECT op.project_uuid, uo.organization_id, "
-            "editor FROM user_organization as uo JOIN organization_project as op"
-            " ON uo.organization_id = op.organization_id "
-            "WHERE user_id = %s) AS op ON p.uuid = op.project_uuid;",
-            [user.id],
-        )
-        org_projects = []
-        for res in project_org_result:
+        projects = []
+        for res in projects_result:
             owner_name = db.execute_return(
                 "SELECT name FROM users WHERE id = %s", [res[2]]
             )
```

Comment:
Why don't we add this to the complex SQL query as well?

Reply:
Yeah we can